lms/static/js/writable_gradebook.js

```diff
@@ -16,26 +16,22 @@ function _templateLoader(templateName, staticPath, callback, errorCallback) {
 }
 
 function courseXblockUpdater(courseID, dataToSend, visibilityData, callback, errorCallback) {
-    var cleanData = {'users' : {}};
-
-    if (dataToSend instanceof Array)
-        for (var i = 0; i < dataToSend.length; i++) {
-            cleanData.users['id_' + dataToSend.userID] = {
-                'block_id' : dataToSend[i].blockID || '',
-                'grade' : dataToSend[i].grade || '',
-                'max_grade' : dataToSend[i].maxGrade || null,
-                'state' :  dataToSend[i].state || '{}',
-                'user_id' : dataToSend[i].userID || ''
-            };
-        }
-    else if (dataToSend instanceof Object)
-        cleanData.users = dataToSend;
-
-    var postUrl = '/api/score/courses/' + courseID;
-
-    if (!_.isEmpty(visibilityData))
-        cleanData.visibility = visibilityData;
+    var cleanData = [];
+    _.each(dataToSend, function(data, _){
```

Comment:
Please use _.map() This way you don't have to push state onto a variable. Instead you can do `var cleanData = _.map(dataToSend, function (data) { return //What the data should look like})`